File: 657-robot-return-to-origin/657-robot-return-to-origin.cpp
class Solution {
public:
    bool judgeCircle(string moves) {
         int u=0,d=0, l=0,r=0;
        
        for(auto x : moves){
            
            if (x=='U') u++;
            
            else if (x=='D') d++;
            
            else if (x=='L') l++;
            
            else r++;
        }
        
        
            return (u!=d || l!=r) ? false : true;
    }
};

File: 657-robot-return-to-origin/657-robot-return-to-origin.java
class Solution {
    public boolean judgeCircle(String moves) {
         int count1 = 0, count2 = 0;
        for(int i=0; i<moves.length(); i++){
            if(moves.charAt(i) == 'R'){
                count1++;
            }else if(moves.charAt(i) == 'U'){
                count2++;
            }else if(moves.charAt(i) == 'L'){
                count1--;
            }else if(moves.charAt(i) == 'D'){
                count2--;
            }
        }
        return (count1 == 0 && count2 == 0);
    }
}